code/modules/atmospherics/machinery/components/unary_devices/vent_pump.dm

```diff
@@ -34,6 +34,18 @@
 	///area this vent is assigned to
 	var/area/assigned_area
 
+	/// The integrity of the 'fan' for this vent; used to control the vent's ability to pump air.
+	var/fan_integrity = 1
```

Comment:
rather than a custom integrity system you should be using the existing atom_integrity